### ISSUE TITLE
Reusable table component

### ISSUE THREAD

**Issue by deboer-tim** [containers/podman-desktop#4365]
### Is your enhancement related to a problem? Please describe

We've discussed future improvements to our 'list pages' like Containers and Pods to be able to:
- be sortable
- allow grouping by container engine/Kubernetes context
- allow the user to pick a different subset of columns, or completely customize the columns or ordering
- click anywhere to go to Details page vs current single column
etc.

Currently, each of these pages has a hardcoded set of columns, and implementing any of these would require changes to every page. As we discuss adding more pages for Kubernetes objects it would be a good time to make sure we don't extend this debt.

### Describe the solution you'd like

Create a basic table component that covers the current column layout requirements of any of these pages. Each page just says 'I have X items and can provide Y sets of columns for each one' and the component is responsible for iterating over the items to create the layout.

For example the list might say it has a name column that should be left justified, a size column that should be right justified, provide 'renderers' for both, and then 5 objects. The component would iterate over these, create the header, trs, tds, and call the renderers at the appropriate places.

This initial component doesn't need to support any of the new features listed, it just needs to allow us to do that in the future in one place. For instance, hiding a column would just mean it leaves out the header and doesn't use the renderer for it.

**Comment by feloy** [comment]
Did we consider using a specialized library, for example AG grid Community (MIT license)? 
https://www.ag-grid.com/
https://ag-grid-svelte.michael.kim/guide/overview/

**Comment by slemeur** [comment]
That's a great suggestion!

**Comment by jeffmaury** [comment]
Good candidate for POC during DevTools week

**Comment by deboer-tim** [comment]
FWIW, I looked at the top >10 tables that came up when I googled variations of 'svelte table component'. Most were very basic, and even the more advanced ones didn't support most of our current requirements (e.g. custom cell rendering w/ actions, custom padding, expandable sections, etc), let alone future requirements like sorting (correctly) or customizing columns, etc.

AG Grid definitely is feature rich and might be an option, but that is a bigger change along with the implications of that - dependencies, applying our UI theme, etc.

@slemeur , @mairin I think we need to make a choice. Continuing down the path of this PR it is totally doable to implement all of the future designs we've seen so far (e.g. sorting, different column sets) but it would continue to be custom and one PR at a time, so there's a limit to how far we'd go. Switching to an existing component would be a bigger initial change and might restrict us in some ways, but hopefully would give us access to a longer list of features. So, what else might we need our tables to do in a year?

We need to make this choice now because we should defer issues from next sprint like grouping by environment if we may replace the table.
